sdk/rust/src/entity_id.rs

```diff
@@ -67,6 +67,19 @@ impl Debug for Checksum {
     }
 }
 
+pub trait AutoValidateChecksum {
```

Comment:
`impl` for `&[T]`?

Reply:
Ah, wait, are you suggesting I should implement `AutoValidateChecksum` for `&[T}` so that I can use that as a shortcut elsewhere in the `validate_checksums_for_ledger_id()` implementations and get rid of the `for` loops?  I think that would only work in a couple of places.  There are a lot of places in the code where I have to iterate over collections of more complex data structures, and this would not help there.  I think I'd rather keep it as-is.

Reply:
that's fine then